soynlp/utils/utils.py

```diff
@@ -97,214 +74,260 @@ def check_corpus(corpus):
+                    line_iterator = tqdm(f, desc='[DoublespaceLineCorpus] iter doc ... ')
             else:
-                f = open(self.corpus_fname, encoding='utf-8')
-        except Exception as e:
-            print(e)
-
-        try:
-            # skip headers
-            for _ in range(self.skip_header):
-                next(f)
-        except Exception as e:
-            print(e)
-
-        # iteration
-        num_sent, stop = 0, False
-        for doc_idx, doc in enumerate(f):
-            if stop:
-                break
-
-            # yield doc
-            if not self.iter_sent:
-                yield doc.strip()
-                if (self.num_doc > 0) and ((doc_idx + 1) >= self.num_doc):
-                    stop = True
-                continue
+                line_iterator = f
 
-            # yield sents
-            for sent in doc.split('  '):
-                if (self.num_sent > 0) and (num_sent >= self.num_sent):
-                    stop = True
+            # iteration
+            num_sent, stop_doc_iter = 0, False
+            for doc_idx, doc in enumerate(line_iterator):
+                if stop_doc_iter:
                     break
-                sent = sent.strip()
-                if sent:
-                    yield sent
-                    num_sent += 1
+                # yield doc
+                if not self.iter_sent:
+                    yield doc.strip()
+                    if (self.num_doc > 0) and ((doc_idx + 1) >= self.num_doc):
+                        stop_doc_iter = True
+                    continue
+                # yield sents
+                for sent in doc.split('  '):
+                    if (self.num_sent > 0) and (num_sent >= self.num_sent):
+                        stop_doc_iter = True
+                        break
+                    sent = sent.strip()
+                    if sent:
+                        yield sent
+                        num_sent += 1
 
     def __len__(self):
-        try:
-            if self.num_doc == 0:
-                self.num_doc, self.num_sent = self._check_length(-1, -1)
-            return self.num_sent if self.iter_sent else self.num_doc
-        except:
-            return -1
+        if self.num_doc == 0:
+            self.num_doc, self.num_sent = self._check_length(-1, -1)
+        return self.num_sent if self.iter_sent else self.num_doc
 
-class EojeolCounter:
-    def __init__(self, sents=None, min_count=1, max_length=15,
-        filtering_checkpoint=0, verbose=False, preprocess=None):
 
+class EojeolCounter:
+    """
+    Args:
+        sents (list of str like) : sentence list
+        min_count (int) : minimum frequency of eojeol
+        max_length (int) : maximum length of eojeol
+        filtering_checkpoints (int) : it drops eojeols which appear less than `min_count` for every `filtering_checkpoints`
+        verbose (Boolean) : if True, it shows progress
+        preprocessing (callable) : sentence preprocessing function
```

Comment:
typo

Reply:
** additional typo not resolved: filtering_checkpoints --> filtering_checkpoint